src/bluesky/utils/__init__.py

```diff
@@ -2010,3 +2010,22 @@ def is_plan(bs_plan):
     """
 
     return inspect.isgeneratorfunction(bs_plan) or getattr(bs_plan, "_is_plan_", False)
+
+
+def deep_update(primary: dict, updates: dict) -> dict:
```

Comment:
```suggestion
def deep_update(primary: Mapping, updates: Mapping) -> Mapping:
```

I think this is what mypy is asking you to do.

Reply:
The original deep_update in Pydantic had

KeyType = TypeVar('KeyType')
def deep_update(mapping: dict[KeyType, Any], *updating_mappings: dict[KeyType, Any]) -> dict[KeyType, Any]:

Although I can't see why it shouldn't be `(Dict[K, V], Dict[K,V]) -> Dict[K,V]`